Integrations/MimecastV2/MimecastV2.py

```diff
@@ -1479,6 +1471,605 @@ def download_attachment_request(attachment_id):
+            'Internal user': str(user['internal'])
+        }
+
+        users_list.append(user_entry)
+
+    md = tableToMarkdown(md, users_list,
+                         ['Name', 'Email address', 'Domain', 'Type', 'Internal user'])
+
+    return md
+
+
+def add_users_under_group_in_context_dict(users_list, group_id):
+    demisto_context = demisto.context()
+
+    if demisto_context and 'Mimecast' in demisto_context:
+        if 'Groups' in demisto_context['Mimecast']:
+            groups_entry_in_context = demisto_context['Mimecast']['Groups']
+            for group in groups_entry_in_context:
+                demisto.results('2')
+                if group['ID'] == group_id:
+                    group['Users'] = users_list
+                    return groups_entry_in_context
+
+    return [
+        {
+            'ID': group_id,
+            'Users': users_list
+        }
+    ]
+
+
+def group_members_api_response_to_context(api_response, group_id=-1):
+    group_id = demisto.args().get('group_id', group_id)
+
+    users_list = list()
+    for user in api_response['data'][0]['groupMembers']:
+        user_entry = {
+            'Name': user['name'],
+            'EmailAddress': user['emailAddress'],
+            'Domain': user['domain'],
+            'Type': user['type'],
+            'InternalUser': str(user['internal']),
+            'IsRemoved': False
+        }
+
+        users_list.append(user_entry)
+
+    groups_after_update = add_users_under_group_in_context_dict(users_list, group_id)
+
+    return {'Mimecast.Groups(val.ID && val.ID == obj.ID)': groups_after_update}
+
+
+def add_remove_member_to_group(action_type):
+    if action_type == 'add':
+        api_endpoint = '/api/directory/add-group-member'
+    else:
+        api_endpoint = '/api/directory/remove-group-member'
+
+    api_response = create_add_remove_group_member_request(api_endpoint)
+
+    markdown_output = add_remove_api_response_to_markdown(api_response, action_type)
+    entry_context = add_remove_api_response_to_context(api_response, action_type)
+
+    return_outputs(markdown_output, entry_context, api_response)
+
+
+def create_add_remove_group_member_request(api_endpoint):
+    group_id = demisto.args().get('group_id', '').encode('utf-8')
+    email = demisto.args().get('email_address', '').encode('utf-8')
+    domain = demisto.args().get('domain_address', '').encode('utf-8')
+
+    data = {
+        'id': group_id,
+    }
+
+    if email:
+        data['emailAddress'] = email
+
+    if domain:
+        data['domain'] = domain
+
+    payload = {
+        'data': [data]
+    }
+
+    response = http_request('POST', api_endpoint, str(payload))
+    if isinstance(response, dict) and response.get('fail'):
+        return_error(json.dumps(response.get('fail', [{}])[0].get('errors')))
+    return response
+
+
+def add_remove_api_response_to_markdown(api_response, action_type):
+    address_modified = api_response['data'][0]['emailAddress']
```

Comment:
will it be email even if we added/removed domain

Reply:
Yes, according to the API documentation.